Core/GDCore/Extensions/Metadata/DependencyMetadata.h

```diff
@@ -0,0 +1,87 @@
+    /**
+     * \brief Contains a information about a Dependency (library) of an extension.
+     * \note Is subject to changes.
+     */
+    class GD_CORE_API DependencyMetadata {
+        public:
+            /**
+             * \brief Sets the name shown to users.
+             */
+            DependencyMetadata& SetName(const gd::String& name_) {
+                name = name_;
+                return *this;
+            };
+
+            /**
+             * \brief Sets the name used by the dependency manager to find the dependency.
+             * \example 
+             *  // For depending upon the NPM package is-thirteen
+             *  gd::DependencyMetadata dependencyMetadata = gd::DependencyMetadata();
+             *  dependencyMetadata.setExporterName("is-thirteen");
+             */
+            DependencyMetadata& SetExporterName(const gd::String& exportName_) {
+                exportName = exportName_;
+                return *this;
+            };
+
+            /**
+             * \brief Set the version of the dependency to install. Use -1 for latest.
+             */
+            DependencyMetadata& SetVersion(const int version_) {
+                version = version_;
+                return *this;
+            };
+
+            /**
+             * \brief Sets the type of dependecy (What will be used to install it)
+             */
+            DependencyMetadata& SetDependencyType(const gd::String& dependencyType_) {
+                dependencyType = dependencyType_;
+                return *this;
+            };
+
+            /**
+             * \brief Sets a dependency type specific setting.
+             */
+            DependencyMetadata& SetExtraSetting(const gd::String& settingName, const gd::PropertyDescriptor settingValue) {
+                extraData[settingName] = std::shared_ptr<gd::PropertyDescriptor>(settingValue);
+                return *this;
+            };
+
+            const gd::String& GetName() const { return name; };
+            const gd::String& GetExportName() const { return exportName; };
+            int GetVersion() const { return version; };
+            DependencyTypes GetDependencyType() const { return dependencyType; };
+            std::shared_ptr<gd::PropertyDescriptor> GetExtraSetting(const gd::String& settingName) const { return extraData[settingName]; };
+
+        private:
+            gd::String name = ""; ///< The name of the dependency.
+            gd::String exportName = ""; ///< The name used to install the package (example: npm package name for npm dependency type).
+            int version = -1; ///< The version of the dependency
+            DependencyTypes dependencyType = DependencyTypes::invalid; ///< The tool used to install the dependency.
+            std::map<gd::String, std::shared_ptr<gd::PropertyDescriptor>> extraData = new std::map<gd::String, std::shared_ptr<gd::PropertyDescriptor>>(); ///< Contains dependency type specific additional parameters for the dependency.
```

Comment:
This usage of new is a bit strange, I'm worried you're instantiating something on the heap. You don't need to do this, objects are initialized with their default constructor when their parent is constructed :)